app/js/components/Login.jsx

```diff
@@ -6,37 +6,73 @@
  * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
  * graphic logo is a trademark of OpenMRS Inc.
  */
-import React from 'react';
+import React, { Component } from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import axios from 'axios';
+import axiosInstance from '../config/axiosInstance';
+import LoginForm from './presentational/LoginForm';
+import  { css } from 'glamor';
 
-import LoginForm from './presentational/LoginForm'
-
-class Login extends React.Component {
+class Login extends Component {
 
   constructor(props) {
     super(props);
     this.state = {
       username: '',
       password: '',
     };
-
-    this.handleChange = this.handleChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this)
   }
 
-  handleChange(event) {
+  handleChange = (event) => {
     this.setState({ [event.target.name] : event.target.value });
   }
 
-
-  handleSubmit(event) {
+  handleSubmit = (event) => {
     event.preventDefault();
     const { username, password } = this.state;
-    this.props.history.push("/")
+    const auth = 'Basic '+ new Buffer(`${username}:${password}`).toString('base64');
+    this.authenticateUser(`/session`, auth)
```

Comment:
Can we really trust the `username` and `password` input from the User, how about you add some user input validations before sending the info to the server?

Reply:
I think having to do regex matching for valid usernames and strong passwords on login is rarely done. To sign up, I totally agree. Secondly, limiting usernames to a certain pattern on the front end must always be in sync with what is on the backend in case someone disables javascript so as to show the same errors. I am not certain what conventions are used in the Java backend for what passes as a valid username. Password validation on login seems like a stretch too.
Take a look at 
- https://web.facebook.com/login.php
- Gmail signIn
- https://www.amazon.com/ap/signin
Validation is hardly done on login at the client side.

Reply:
Definitely not. I agree with you.  I have added empty form field validation in case a user edits the HTML:5 required attribute. Usernames take on many forms unless agreed upon and passwords are not validated on login.
- https://www.amazon.com/ap/signin
- Gmail sign in
- https://web.facebook.com/login.php
I guess it is because the user input is not stored but used in comparison with what was stored at sign up.